compiler/src/dmd/dtemplate.d

```diff
@@ -990,40 +990,47 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
         return buf.extractChars();
     }
 
-    private void formatParamsWithTiargs(ref Objects tiargs, ref OutBuffer buf)
+    /*******************************************
+     * Append to buf a textual representation of template parameters with their arguments.
+     * Params:
+     *  parameters = the template parameters
+     *  tiargs = the correspondeing template arguments
+     *  variadic = if it's a variadic argument list
+     *  buf = where the text output goes
+     */
+    static private void formatParamsWithTiargs(ref TemplateParameters parameters, ref Objects tiargs, bool variadic, ref OutBuffer buf)
     {
         buf.writestring("  with `");
 
         // write usual arguments line-by-line
         // skips trailing default ones - they are not present in `tiargs`
-        const bool variadic = isVariadic() !is null;
-        const end = cast(int)parameters.length - (variadic ? 1 : 0);
-        uint i;
+        const end = parameters.length - (variadic ? 1 : 0);
+        size_t i;
         for (; i < tiargs.length && i < end; i++)
         {
-            if (i > 0)
+            if (i)
```

Comment:
I think `if (i > 0)` reads better

Reply:
since `i` is now unsigned, the `> 0` becomes misleading (although still correct).